File: android/ValleTPV/app/src/main/java/com/valleapp/valletpv/dlg/DlgMensajes.java
package com.valleapp.valletpv.dlg;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;

import android.widget.ImageButton;
import android.widget.ListView;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.valleapp.valletpv.R;
import com.valleapp.valletpv.adaptadoresDatos.AdaptadorCamMensajes;
import com.valleapp.valletpv.interfaces.IControlMensajes;

import org.json.JSONObject;

import java.util.List;

public class DlgMensajes extends Dialog implements IControlMensajes {

    private final IControlMensajes controlador;
    final Context cx;

    public DlgMensajes(@NonNull Context context, IControlMensajes controlMensajes) {
        super(context);
        this.controlador = controlMensajes;
        this.cx = context;
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_mensaje);
        ImageButton btn = findViewById(R.id.btn_salir_mensajes);
        btn.setOnClickListener(v -> cancel());
        setTitle("Enviar mensajes");
    }

    @Override
    public void sendMensaje(String IDRecptor, String mensaje) {
        TextView t = findViewById(R.id.txt_mensaje);
        if (!t.getText().toString().equals("")) {
            this.controlador.sendMensaje(IDRecptor, t.getText().toString());
            cancel();
        }
    }

    public void mostrarReceptores(List<JSONObject> lista){
        ListView l = findViewById(R.id.lista_camareros_notificables);
        l.setAdapter(new AdaptadorCamMensajes(cx, lista, this));
    }


}

File: android/ValleTPV/ValleTPV/src/main/java/com/valleapp/valletpv/dlg/DlgMensajes.kt
package com.valleapp.valletpv.dlg

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.widget.ImageButton
import android.widget.ListView
import android.widget.TextView
import com.valleapp.valletpv.R
import com.valleapp.valletpv.adaptadoresDatos.AdaptadorCamMensajes
import com.valleapp.valletpv.interfaces.IControlMensajes
import org.json.JSONObject

class DlgMensajes(
    context: Context,
    private val controlador: IControlMensajes
) : Dialog(context), IControlMensajes {

    private val cx: Context = context

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dialog_mensaje)
        val btn: ImageButton = findViewById(R.id.btn_salir_mensajes)
        btn.setOnClickListener { cancel() }
        setTitle("Enviar mensajes")
    }

    override fun sendMensaje(IDRecptor: String, mensaje: String) {
        val t: TextView = findViewById(R.id.txt_mensaje)
        if (t.text.toString().isNotEmpty()) {
            controlador.sendMensaje(IDRecptor, t.text.toString())
            cancel()
        }
    }

    fun mostrarReceptores(lista: List<JSONObject>) {
        val l: ListView = findViewById(R.id.lista_camareros_notificables)
        l.adapter = AdaptadorCamMensajes(cx, lista, this)
    }
}
